GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -529,12 +532,25 @@ public FileSystemTaskResult AddMissingModifiedFiles()
             {
                 if (this.modifiedFilesInvalid)
                 {
-                    FileSystemTaskResult result = this.indexParser.AddMissingModifiedFiles(this.context.Tracer, this.indexFileStream);
+                    Dictionary<string, PlaceholderListDatabase.PlaceholderData> placeholderDataByPath;
+                    List<PlaceholderListDatabase.PlaceholderData> placeholdersToKeep;
+                    this.placeholderList.GetAllEntries(out placeholderDataByPath, folderPlaceholders: out placeholdersToKeep);
+
+                    FileSystemTaskResult result = this.indexParser.AddMissingModifiedFiles(this.context.Tracer, this.indexFileStream, placeholderDataByPath, placeholdersToKeep);
                     if (result == FileSystemTaskResult.Success)
                     {
                         this.modifiedFilesInvalid = false;
                     }
 
+                    // Any paths that are not in the the index must be added to ModifiedPaths.dat so that git
+                    // reports them as untracked
+                    foreach (string path in placeholderDataByPath.Keys)
+                    {
+                        this.AddModifiedPath(path);

```

Comment:
`this.AddModifiedPath(path)` can return an error and we need to handle it (also rename `AddModifiedPath` to `TryAddModifiedPath`)

Reply:
Most of the `Try` methods return a `bool` to check where this is returning a `FileSystemTaskResult` so I'm not sure it fits how we are using TryX methods.

Reply:
That's true.  After taking a look at the other methods in this class it looks like we're not using `Try` for those that can fail and return `FileSystemTaskResult`.  I'll just update the code to check the return value.